src/EnergyPlus/HVACManager.cc

```diff
@@ -2725,6 +2730,46 @@ namespace HVACManager {
 
 	}
 
+	void
+	CheckMassBalances()
+	{
+		using namespace Fans;
+		using namespace Pumps;
+		using General::RoundSigDigits;
+		using DataHVACGlobals::SmallMassFlow;  // Air
+		using DataBranchAirLoopPlant::MassFlowTolerance;  // Liquid
+
+		for (int FanNum = 1; FanNum <= NumFans; ++FanNum) {
+
+//			if (abs(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate - Fan( FanNum ).InletAirMassFlowRate) > SmallMassFlow) {
+//				ShowFatalError(Fan(FanNum).FanType + ", Name: " + Fan(FanNum).FanName + ": Fan inlet mass flow rate (" + RoundSigDigits(Fan( FanNum ).InletAirMassFlowRate, 4) + ") does not match inlet node mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate, 4) + ").");
+//			}
+//
+//			if (abs(Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate - Fan( FanNum ).OutletAirMassFlowRate) > SmallMassFlow) {
+//				ShowFatalError(Fan(FanNum).FanType + ", Name: " + Fan(FanNum).FanName + ": Fan outlet mass flow rate (" + RoundSigDigits(Fan( FanNum ).OutletAirMassFlowRate, 4) + ") does not match outlet node mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate, 4) + ").");
+//			}
+
+			if (abs(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate - Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate) > SmallMassFlow) {
+//				ShowFatalError(Fan(FanNum).FanType + ", Name: " + Fan(FanNum).FanName + ": Fan outlet mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate, 4) + ") does not match fan inlet node mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate, 4) + ").");
+// can we get a handle on why this is happening before we start crashing user files?
```

Comment:
Cycling fan is handled in the parent. The flow entering the fan is the average over the time step. The fan is not aware of cycling fan mode except in the OnOffFanPartLoadFraction variable used to adjust fan power.